File: labrabota9-gr15-alekseykrazhev/example2/example2/ViewController.m
//
//  ViewController.m
//  example2
//
//  Created by Кражевский Алексей И. on 6/6/22.
//

#import "ViewController.h"

@interface ViewController ()
@property (strong, nonatomic) IBOutlet UIView *regV;
@property (strong, nonatomic) IBOutlet UISegmentedControl *segm;
@property (strong, nonatomic) IBOutlet UITextField *loginL;
@property (strong, nonatomic) IBOutlet UITextField *passL;
@property (strong, nonatomic) IBOutlet UITextField *loginR;
@property (strong, nonatomic) IBOutlet UITextField *mailR;
@property (strong, nonatomic) IBOutlet UITextField *passR;
@property (strong, nonatomic) IBOutlet UITextField *r_passR;
@property (strong, nonatomic) IBOutlet UISwitch *agreeR;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [_regV setHidden:YES];
    NSUserDefaults *user = [NSUserDefaults standardUserDefaults];
    
    if ([user objectForKey:@"login"] != nil || [user objectForKey:@"pass"] !=nil) {
        [self performSegueWithIdentifier:@"closeZone" sender:self];
    }
}

- (IBAction)segment:(id)sender {
    UISegmentedControl *seg = (UISegmentedControl *)sender;
    NSInteger select = seg.selectedSegmentIndex;
    
    if (select == 0){
        [_regV setHidden:YES];
    } else {
        [_regV setHidden:NO];
    }
}
- (IBAction)segmentR:(id)sender {
    UISegmentedControl *seg = (UISegmentedControl *)sender;
    NSInteger select = seg.selectedSegmentIndex;
    
    if (select == 0){
        [_regV setHidden:NO];
    } else {
        [_regV setHidden:YES];
    }
}

- (IBAction)buttonL:(id)sender {
    if (![_loginL.text isEqualToString:@""] || ![_passL.text isEqualToString:@""]) {
        NSUserDefaults *user = [NSUserDefaults standardUserDefaults];
        [user setObject:_loginL.text forKey:@"login"];
        [user setObject:_passL.text forKey:@"pass"];
        [self performSegueWithIdentifier:@"closeZone" sender:self];
    }
}

- (IBAction)buttonR:(id)sender {
    if (![_loginR.text isEqualToString:@""] || ![_passR.text isEqualToString:@""] || ![_mailR.text isEqualToString:@""] || ![_r_passR.text isEqualToString:@""]) {
        
        if (![_passR.text isEqualToString:_r_passR.text]){
            NSLog(@"passwords don't match");
            return;
        }
        
        if (!_agreeR.on){
            NSLog(@"agree with terms!");
            return;
        }
        
        NSUserDefaults *user = [NSUserDefaults standardUserDefaults];
        [user setObject:_loginL.text forKey:@"login"];
        [user setObject:_passL.text forKey:@"pass"];
        [self performSegueWithIdentifier:@"closeZone" sender:self];
    } else {
        NSLog(@"fill the rest!");
    }
}

@end

File: labrabota9-gr15-alekseykrazhev/task2-1/task2-1/ViewController.swift
//
//  ViewController.swift
//  task2-1
//
//  Created by Кражевский Алексей И. on 6/8/22.
//

import UIKit

class ViewController: UIViewController{

    @IBOutlet weak var loginL: UITextField!
    @IBOutlet weak var passL: UITextField!
    @IBOutlet weak var signUpView: UIView!
    @IBOutlet weak var loginR: UITextField!
    @IBOutlet weak var mailR: UITextField!
    @IBOutlet weak var passR: UITextField!
    @IBOutlet weak var c_passR: UITextField!
    @IBOutlet weak var errorR: UILabel!
    @IBOutlet weak var errorL: UILabel!
    @IBOutlet weak var terms: UISwitch!
    var isLoggedIn: Bool = false
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        signUpView.isHidden = true
        isLoggedIn = false
    }

    @IBAction func signIn(_ sender: Any) {
        if (loginL.text!.isEmpty || passL.text!.isEmpty) {
            errorL.text = "Smth is missing..."
            return
        }
        UserDefaults.standard.set(loginL.text, forKey: "login")
        UserDefaults.standard.set(passL.text, forKey: "pass")
        isLoggedIn = true
        performSegue(withIdentifier: "toParks", sender: self)
        //self.reloadInputViews()
    }
    
    @IBAction func signUp(_ sender: Any) {
        signUpView.isHidden = false
        errorL.isHidden = true
    }
    
    @IBAction func signInR(_ sender: Any) {
        signUpView.isHidden = true
        errorL.isHidden = false
        
    }
    
    @IBAction func signUpR(_ sender: Any) {
        if (loginR.text!.isEmpty || passR.text!.isEmpty || mailR.text!.isEmpty || c_passR.text!.isEmpty) {
            errorR.text = "Some info is missing!"
            return
        }
        
        if (passR.text != c_passR.text) {
            errorL.text = "Passwords don't match!"
            return
        }
        
        if ((terms == nil)){
            errorL.text = "Agree with terms!"
            return
        }
        
        UserDefaults.standard.set(loginL.text, forKey: "login")
        UserDefaults.standard.set(passL.text, forKey: "pass")
        UserDefaults.standard.set(mailR.text, forKey: "mail")
        isLoggedIn = true
        performSegue(withIdentifier: "toParks", sender: self)
    }

    
}


class ParkCell: UICollectionViewCell{
    
    @IBOutlet weak var name: UILabel!
    
    
}


class NewViewController: UIViewController, UICollectionViewDataSource, UICollectionViewDelegate{
    
    var list = NSDictionary();
    @IBOutlet var pictureField: UIImageView!
    @IBOutlet var descriptionField: UITextView!
    
    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
        list.count;
    }
    
    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
        let cell = collectionView
            .dequeueReusableCell(withReuseIdentifier: "MyCell", for: indexPath) as! ParkCell
        
        cell.name.text=((list.allKeys ) [indexPath.row] as! String);
  
        return cell
    }
    
    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
       
        let courseInfo = (list.object(forKey: ((list.allKeys ) [indexPath.row]))) as! NSDictionary;
        
        //NSLog(courseInfo.description)
       
        descriptionField.text = NSLocalizedString(courseInfo.object(forKey: "Description") as! String, comment: "Some comment");
        let image: UIImage = UIImage(named: courseInfo.object(forKey: "Icon") as! String)!
        pictureField = UIImageView(image: image)
        pictureField.frame = CGRect(x: 0, y: 600, width: 500, height: 252)
        view.addSubview(pictureField)
   }
    
    override func viewDidLoad() {
        super.viewDidLoad()
    
        let filePath = Bundle.main.path(forResource: "myData", ofType: "plist")
        list = NSDictionary(contentsOfFile:filePath!)!
        //NSLog(list.description)
    }
    
}
